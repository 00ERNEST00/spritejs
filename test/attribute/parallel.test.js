import ParallelAttr from '../../src/attribute/parallel';

describe('attribute set get', () => {
  const subject = {
    onPropertyChange(key, value, oldValue, attributes) { },
  };
  const attr = new ParallelAttr(subject);

  test('get set', () => {
    expect(attr.d).toBe('');
    attr.sides = [10, 20];
    expect(attr.d).toMatchSnapshot();

    let angle = Math.PI / 2;
    let x1 = 20 * Math.cos(angle);
    let y1 = 20 * Math.sin(angle);
    expect(attr.points).toEqual([0, 0, 10, 0, x1 + 10, y1, x1, y1]);

    attr.angle = 60;
    angle = Math.PI / 3;
    x1 = 20 * Math.cos(angle);
    y1 = 20 * Math.sin(angle);
    expect(attr.points).toEqual([0, 0, 10, 0, x1 + 10, y1, x1, y1]);

    attr.sides = [0, 0];
    expect(attr.d).toBe('M0 0L0 0L0 0L0 0Z');
  });
});