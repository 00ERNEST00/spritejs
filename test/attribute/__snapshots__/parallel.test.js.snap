// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`attribute set get get set 1`] = `"M0 0L10 0L10.000000000000002 20L1.2246467991473533e-15 20Z"`;
